app/admin/js_infos.rb

```diff
@@ -1,18 +1,82 @@
 ActiveAdmin.register JsInfo do
 
-  # See permitted parameters documentation:
-  # https://github.com/activeadmin/activeadmin/blob/master/docs/2-resource-customization.md#setting-up-strong-parameters
-  #
-  # Uncomment all parameters which should be permitted for assignment
-  #
-  # permit_params :js_name, :url_id, :status, :version, :other_data
-  #
-  # or
-  #
-  # permit_params do
-  #   permitted = [:js_name, :url_id, :status, :version, :other_data]
-  #   permitted << :other if params[:action] == 'create' && current_user.admin?
-  #   permitted
-  # end
-  
+  actions :index, :show
+  filter :url_id
+  filter :status , :as => :select, :collection => JsInfo::STATUS.invert
+  filter :first_seen
+  filter :last_seen
+  filter :created_at
+  filter :updated_at
+
+  index do 
+    id_column
+    column :js_name
+    column 'Version' do |js_info|
+      if js_info.version == JsInfo::Version::NOTFOUND
+        div ("Not found"), :style => 'color : red'
+      else 
+        js_info.version
+      end
+    end
+    column "Usage" do |js|
+      if !params["test_id"]
+        url_ids = JsInfo.where(:js_name => js.js_name, :status => JsInfo::Status::ACTIVE).pluck(:url_id)
+        link_to "#{url_ids.count} :: urls", admin_urls_path('q[id_in]' => url_ids)
+      else
+        url_ids = JsInfo.where("first_seen <= ?", params["test_id"]).
+          where("last_seen >= ?",params["test_id"]).
+          where(:js_name => js.js_name).pluck(:url_id)
+        link_to "#{url_ids.count} :: urls", admin_urls_path("q[id_in]" => url_ids)
+      end
+    end
+    column :first_seen
+    column :last_seen
+    if params[:q] and params[:q][:url_id_equals]
```

Comment:
what's q?

Reply:
q is a query parameter which comes with every query